spec/rationals.nim

```diff
@@ -0,0 +1,193 @@
+## Provides a simple implementation for rational numbers and operations
+## on them.
+
+import
+  bignums,
+  std/math
+
+type Rational* = object
+  ## A rational number represented as a fraction of two arbitrary-precision
+  ## integers.
+  num, den: Bignum
+
+const
+  One = 1'n
+
+proc reduced(r: sink Rational): Rational =
```

Comment:
Any reason behind the preference for the past tense?

`myRational.reduced` vs `myRational.reduce`... they're not super different, I don't see the need to change it, just curious

Reply:
> Any reason behind the preference for the past tense?

Yep, it's because I'm using the naming convention from Nim/NimSkull, where routines that modify their inputs in-place (via a `var` parameter) use the imperative, whereas routines that don't use past tense.

`reduced` could also reduce the rational in-place (in which case the procedure would be named `reduce`), but then all callsites would need to use either temporaries or the result variable, which is longer, and I also have a slight preference towards a more functional style.